src/ecmascript/spidermonkey/document.c

```diff
@@ -450,25 +478,57 @@ document_write_do(JSContext *ctx, unsigned int argc, JS::Value *rval, int newlin
 	struct string *ret = interpreter->ret;
 	JS::CallArgs args = JS::CallArgsFromVp(argc, rval);
 
-	if (argc >= 1 && ret) {
-		int i = 0;
+	char *code;
 
-		for (; i < argc; ++i) {
-			char *code = jsval_to_string(ctx, args[i]);
+	code = stracpy("");
 
-			add_to_string(ret, code);
-		}
+	int numeric_arg;
+
+	if (argc >= 1)
+	{
+		for (int i=0;i<argc;++i) 
+		{
 
-		if (newline)
-			add_char_to_string(ret, '\n');
+			code = jshandle_value_to_char_string(ctx,&args[i]);
+		}
+	
+		if (newline) 
+		{
+			add_to_strn(&code, "\n");
+		}
 	}
+	//DBG("%s",code);
+
 	/* XXX: I don't know about you, but I have *ENOUGH* of those 'Undefined
 	 * function' errors, I want to see just the useful ones. So just
 	 * lighting a led and going away, no muss, no fuss. --pasky */
 	/* TODO: Perhaps we can introduce ecmascript.error_report_unsupported
 	 * -> "Show information about the document using some valid,
 	 *  nevertheless unsupported methods/properties." --pasky too */
 
+	struct document_view *doc_view = interpreter->vs->doc_view;
+	struct document *document;
+	document = doc_view->document;
+	struct cache_entry *cached = doc_view->document->cached;
+	struct fragment *f = cached ? cached->frag.next : NULL;
```

Comment:
This inittialzation is not needed, because, a few lines below you override f.
struct fragment *f; is sufficient

Reply:
O.K., agreed. Fixed.